go/libraries/doltcore/env/remotes.go

```diff
@@ -360,45 +360,50 @@ type PullSpec struct {
 	Branch     ref.DoltRef
 }
 
-func NewPullSpec(ctx context.Context, rsr RepoStateReader, remoteName string, squash, noff, force, remoteOnly bool) (*PullSpec, error) {
-	branch := rsr.CWBHeadRef()
-
+func NewPullSpec(_ context.Context, rsr RepoStateReader, remoteName, remoteRefName string, squash, noff, force, remoteOnly bool) (*PullSpec, error) {
 	refSpecs, err := GetRefSpecs(rsr, remoteName)
 	if err != nil {
 		return nil, err
 	}
-
 	if len(refSpecs) == 0 {
 		return nil, ErrNoRefSpecForRemote
 	}
 
-	trackedBranches, err := rsr.GetBranches()
+	remotes, err := rsr.GetRemotes()
 	if err != nil {
 		return nil, err
 	}
+	remote := remotes[refSpecs[0].GetRemote()]
 
-	trackedBranch, hasUpstream := trackedBranches[branch.GetPath()]
-	if !hasUpstream {
-		if remoteOnly {
-			return nil, fmt.Errorf(ErrPullWithRemoteNoUpstream.Error(), remoteName)
-		} else {
-			return nil, ErrPullWithNoRemoteAndNoUpstream
+	var remoteRef ref.DoltRef
+	if remoteRefName == "" {
+		branch := rsr.CWBHeadRef()
+		trackedBranches, err := rsr.GetBranches()
+		if err != nil {
+			return nil, err
 		}
-	}
 
-	remotes, err := rsr.GetRemotes()
-	if err != nil {
-		return nil, err
+		trackedBranch, hasUpstream := trackedBranches[branch.GetPath()]
+		if !hasUpstream {
+			if remoteOnly {
+				return nil, fmt.Errorf(ErrPullWithRemoteNoUpstream.Error(), remoteName)
```

Comment:
i probably wrote the wrong version, but i think this way works?
```suggestion
				return nil, fmt.Errorf("%w: '%s'", ErrPullWithRemoteNoUpstream, remoteName)
```

Reply:
I see what you mean... this line looks a little odd... I believe it's still correct though, because the error string already contains the `%s` to plug the remote name into it. It would be nice to switch over to `go-errors.v1` and then this could be cleaned up a little more, but I think that's a bigger change (to move all the errors in this file over to go-errors.v1) than is needed right now.